modules/javafx.controls/src/test/java/test/javafx/scene/control/TableViewHorizontalArrowsTest.java

```diff
@@ -0,0 +1,226 @@
+    private TableColumn<String, String> col1;
+    private TableColumn<String, String> col2;
+    private TableColumn<String, String> col3;
+    private TableColumn<String, String> col4;
+
+    private KeyEventFirer keyboard;
+    private StageLoader stageLoader;
+    private NodeOrientation orientation;
+
+    public TableViewHorizontalArrowsTest(NodeOrientation val) {
+        orientation = val;
+    }
+
+    @Before
+    public void setup() {
+        tableView = new TableView<String>();
+        tableView.setNodeOrientation(orientation);
+        sm = tableView.getSelectionModel();
+        fm = tableView.getFocusModel();
+
+        sm.setSelectionMode(SelectionMode.MULTIPLE);
+        sm.setCellSelectionEnabled(false);
+
+        tableView.getItems().setAll("1", "2", "3", "4", "5", "6", "7", "8", "9",
+                                    "10", "11", "12");
+
+        col0 = new TableColumn<String, String>("col0");
+        col1 = new TableColumn<String, String>("col1");
+        col2 = new TableColumn<String, String>("col2");
+        col3 = new TableColumn<String, String>("col3");
+        col4 = new TableColumn<String, String>("col4");
+        tableView.getColumns().setAll(col0, col1, col2, col3, col4);
+
+        keyboard = new KeyEventFirer(tableView);
+
+        stageLoader = new StageLoader(tableView);
+        stageLoader.getStage().show();
+    }
+
+    @After
+    public void tearDown() {
+        tableView.getSkin().dispose();
+        stageLoader.dispose();
+    }
+
+    // ---------------- Helper methods -------------------------
+    /**
+     * Toggles the nodeOrientation of tableView.
+     */
+    private void changeNodeOrientation() {
+        orientation = (orientation == NodeOrientation.LEFT_TO_RIGHT?
+            NodeOrientation.RIGHT_TO_LEFT : NodeOrientation.LEFT_TO_RIGHT);
+        tableView.setNodeOrientation(orientation);
+    }
+
+    /**
+     * Orientation-aware forward horizontal navigation with arrow keys.
+     * @param modifiers the modifiers to use on keyboard
+     */
+    private void forward(KeyModifier... modifiers) {
+        if (orientation == NodeOrientation.LEFT_TO_RIGHT) {
+            keyboard.doRightArrowPress(modifiers);
+        } else if (orientation == NodeOrientation.RIGHT_TO_LEFT) {
+            keyboard.doLeftArrowPress(modifiers);
+        }
+    }
+
+    /**
+     * Orientation-aware backward horizontal navigation with arrow keys.
+     * @param modifiers the modifiers to use on keyboard
+     */
+    private void backward(KeyModifier... modifiers) {
+        if (orientation == NodeOrientation.LEFT_TO_RIGHT) {
+            keyboard.doLeftArrowPress(modifiers);
+        } else if (orientation == NodeOrientation.RIGHT_TO_LEFT) {
+            keyboard.doRightArrowPress(modifiers);
+        }
+    }
+
+
+    // ----------------------- Tests ----------------------------
+
+    @Test
+    public void testForwardSelect() {
+        sm.setCellSelectionEnabled(true);
```

Comment:
here (and in other test methods) not needed if cellSelection is enabled in setup